cmake/tests/cxx20_lambda_capture.cpp

```diff
@@ -0,0 +1,20 @@
+//  Copyright (c) 2021 Srinivas Yadav
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// test for availability of Allow lambda-capture [=, this] (C++20)
+
+struct S
+{
+    void f()
+    {
+        [=, this]{};
```

Comment:
@srinivasyadav18 please make sure that the test compiles without any other warnings as well. The test seems to be failing even with C++20 because this lambda is unused. Something like this should silence the warning, while keeping the warning related to the capture of `this`: `(void)[=, this]{}`.

Reply:
@msimberg I have made the changes, but can you please share the link where it is failing, it will be more clear for me. :).
Thank You.

Reply:
This is on the clang-newest configuration (which is clang 11 with C++20): https://cdash.cscs.ch/viewConfigure.php?buildid=159985. Note the failing `HPX_WITH_CXX20_LAMBDA_CAPTURE` test. Interestingly with gcc the test does actually pass, so I suppose it for some reason does not consider that an unused variable (at least with the default warnings).